pos_margin/static/src/js/models.js

```diff
@@ -0,0 +1,56 @@
+// Copyright (C) 2021 - Today: GRAP (http://www.grap.coop)
+// @author: Sylvain LE GAL (https://twitter.com/legalsylvain)
+// License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+odoo.define("pos_margin.models", function (require) {
+    "use strict";
+
+    var models = require("point_of_sale.models");
+
+    // /////////////////////////////
+    // Overload models.Order
+    // /////////////////////////////
+    var OrderMargin = models.Order.extend({
+
+        get_margin: function() {
+            var margin = 0;
+            this.orderlines.each(function(orderline){
+                margin += orderline.get_margin();
+            })
```

Comment:
```suggestion
            return this.get_orderlines().reduce(
                (margin, line) => margin += line.get_margin(), 0
            )
```

This achieves the same
not blocking though